ingest/expression_files/dense_ingestor.py

```diff
@@ -40,21 +40,36 @@ def __init__(self, file_path, study_file_id, study_id, **matrix_kwargs):
         self.gene_names = {}
         self.header = DenseIngestor.process_header(next(self.csv_file_handler))
 
+    def matches_file_type(file_type):
+        return file_type == "dense"
+
     def execute_ingest(self):
         # Method can only be executed once due to
         # dependency on position in text file.
         # Row after header is needed for R format validation
         row = next(self.csv_file_handler)
-        if not DenseIngestor.is_valid_format(self.header, row):
+        if not DenseIngestor.is_valid(
+            self.header,
+            row,
+            query_params=(self.study_id, self.mongo_connection._client),
+        ):
             raise ValueError("Dense matrix has invalid format")
```

Comment:
Is there any way for the user to know what the specific issue with the file was?  Or will they get the same error message for both a malformed header and a duplicate gene name?

Reply:
Tracked [here](https://broadworkbench.atlassian.net/browse/SCP-2660)